src/display/canvas.js

```diff
@@ -913,6 +913,14 @@ function composeSMask(ctx, smask, layerCtx, layerBox) {
 }
 
 function getImageSmoothingEnabled(transform, interpolate) {
+  // In section 8.9.5.3 of the PDF spec, it's mentioned that the interpolate
+  // flag should be used when the image is upscaled.
+  // In Firefox, smoothing is always used when downscaling images (bug 1360415).
```

Comment:
I'm not entire why, but this change does still seem to have an influence on Firefox. Notice how on http://54.193.163.58:8877/78e3c81b81dcbab/reftest-analyzer.html the `firefox-bug956965-page33` tests seems to be more "blocky" now (on both Windows and Linux, and in Firefox and Chrome by the way). I went over all other changed reference tests and all of them seem like an improvement, so this particular one isn't a blocker for me, but it is a bit odd that this one seems to render a bit less nice than before so I figured I'd ask.

Reply:
It's because here we're considering that we downscale when the scale factor is lower than deviceRatio x 96/72 when in Firefox it's when the factor is lower than 1.
So we're taking into account the resolution.

Reply:
And about `firefox-bug956965-page33`, I was surprised too but in checking on a non-hdpi screen (an old one I got almost 10y ago), there is no perceptible differences.

Reply:
Thank you for the explanation; in that case this sounds good to me and doesn't require further changes.